crypto/src/main/java/org/springframework/security/crypto/bcrypt/BCryptPasswordEncoder.java

```diff
@@ -119,6 +119,11 @@ public boolean matches(CharSequence rawPassword, String encodedPassword) {
 			return false;
 		}
 
+		if(rawPassword.toString() == null || rawPassword.toString.length() == 0){
```

Comment:
Let's have this match the earlier check:

```suggestion
		if (rawPassword == null || rawPassword.length() == 0) {
```

Also, for readability, we should probably place this at the beginning of the method, so that the method parameters are testing in the same order that they are defined.